src/nifs/protogalaxy/pow_i.rs

```diff
@@ -0,0 +1,123 @@
+use std::{iter, num::NonZeroUsize};
+
+use bitter::{BitReader, LittleEndianReader};
+use ff::PrimeField;
+use tracing::*;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Input `t` is not enought for represent `i`")]
+    TooLittleT {
+        t: NonZeroUsize,
+        representation_lenght: NonZeroUsize,
+    },
+}
+
+fn pow_i<F: PrimeField>(
+    i: usize,
+    t: NonZeroUsize,
+    challenges_powers: impl Iterator<Item = F>,
+) -> F {
+    let bytes = i.to_le_bytes();
```

Comment:
add assertion of `i<2^t`

Reply:
I don't check anything meaningfully here, as this is an internal function that will be run on many threads. All the checks are put into the public fn